src/backend/distributed/utils/resource_lock.c

```diff
@@ -1081,14 +1083,21 @@ EnsureCanAcquireLock(Oid relationId, LOCKMODE lockMode)
 
 
 /*
- * AcquireDistributedLockOnRelations acquire a distributed lock on worker nodes
+ * AcquireDistributedLockOnRelations_Internal acquire a distributed lock on worker nodes
  * for given list of relations ids. Worker node list is sorted so that the lock
- * is acquired in the same order regardless of which node it was run on. Notice
- * that no lock is acquired on coordinator node if the coordinator is not added
- * to the metadata.
+ * is acquired in the same order regardless of which node it was run on.
+ * Notice that no lock is acquired on coordinator node if the coordinator is not
+ * added to the metadata.
+ *
+ * Notice that no validation is done on the relationIds, that is the responsibility of
```

Comment:
AcquireDistributedLockOnRelations (or any caller in general)

Reply:
I will generalize this to specify any caller.